# HG changeset patch
# User Sean Farley <sean@mcs.anl.gov>
# Date 1330211528 21600
# Node ID b8b0cc4abb9bed53f909e0ac38b29d9f76875236
# Parent  f6fc32c3dec2c9875d6f6412039542111cb1e3e7
metis: fix vertex weights if not compressed

diff --git a/src/cmumps_part2.F b/src/cmumps_part2.F
--- a/src/cmumps_part2.F
+++ b/src/cmumps_part2.F
@@ -1437,10 +1437,14 @@
                FILS(I) = 2
             ENDDO
             DO I=KEEP(93)/2+1,NCMP
                FILS(I) = 1
             ENDDO
+         ELSE
+            DO I=1,NCMP
+               FILS(I) = 1
+            ENDDO
          ENDIF
          CALL METIS_NODEND(NCMP, IW(L2), IW(1),FILS,
      &        OPTIONS_METIS,
      &        IKEEP(1,2), IKEEP(1,1) )
          DEALLOCATE (OPTIONS_METIS)
diff --git a/src/dmumps_part2.F b/src/dmumps_part2.F
--- a/src/dmumps_part2.F
+++ b/src/dmumps_part2.F
@@ -1433,10 +1433,14 @@
                FILS(I) = 2
             ENDDO
             DO I=KEEP(93)/2+1,NCMP
                FILS(I) = 1
             ENDDO
+         ELSE
+            DO I=1,NCMP
+               FILS(I) = 1
+            ENDDO
          ENDIF
          CALL METIS_NODEND(NCMP, IW(L2), IW(1),FILS,
      &        OPTIONS_METIS,
      &        IKEEP(1,2), IKEEP(1,1) )
          DEALLOCATE (OPTIONS_METIS)
diff --git a/src/smumps_part2.F b/src/smumps_part2.F
--- a/src/smumps_part2.F
+++ b/src/smumps_part2.F
@@ -1433,10 +1433,14 @@
                FILS(I) = 2
             ENDDO
             DO I=KEEP(93)/2+1,NCMP
                FILS(I) = 1
             ENDDO
+         ELSE
+            DO I=1,NCMP
+               FILS(I) = 1
+            ENDDO
          ENDIF
          CALL METIS_NODEND(NCMP, IW(L2), IW(1),FILS,
      &        OPTIONS_METIS,
      &        IKEEP(1,2), IKEEP(1,1) )
          DEALLOCATE (OPTIONS_METIS)
diff --git a/src/zmumps_part2.F b/src/zmumps_part2.F
--- a/src/zmumps_part2.F
+++ b/src/zmumps_part2.F
@@ -1437,10 +1437,14 @@
                FILS(I) = 2
             ENDDO
             DO I=KEEP(93)/2+1,NCMP
                FILS(I) = 1
             ENDDO
+         ELSE
+            DO I=1,NCMP
+               FILS(I) = 1
+            ENDDO
          ENDIF
          CALL METIS_NODEND(NCMP, IW(L2), IW(1),FILS,
      &        OPTIONS_METIS,
      &        IKEEP(1,2), IKEEP(1,1) )
          DEALLOCATE (OPTIONS_METIS)
